### PR TITLE
Add tutorial for LOJ-1011 Marriage Ceremonies

### PR DESCRIPTION
**Pull request by nfpranta**

### CODE REVIEW

**Comment by rebornplusplus** [comment]
Hello @nfpranta, good work. But I think there's some room to improve here.

Please insert texts in backticks(`) when they are meant to be code-ish. For "1st", "2nd" or "this testcase" etc. we can simply not use a backtick at all. When you need to emphasis on a phrase however, say "Bitmask function", use quotation marks or make it italic or bold.

When explaining a DP solution, if you are defining a function, define the transitions, corner cases and base cases properly. It's exhausting to read a paragraph full of text to parse the definition. For example, if someone is to define `fib(x)` to be the x'th Fibonacci number, (s)he might insert the following:
```
fib(0) = 0
fib(1) = 1
fib(x) = fib(x - 1) + fib(x - 2)   ; for x > 1
```

Also insert three backticks consecutively when you want to display a block of code, in your case, the grid from the second test case.

Chop the comments in your code so that they can be viewed without scrolling horizontally. You might keep a right margin of say, 70 to accomplish that.

**Comment by nfpranta** [comment]
Can you give the site which you gave last time to check how it actually looks on the screen?

**Comment by Tahanima** [comment]
@nfpranta I think [this](https://dillinger.io/) is what you are looking for.

**Comment by nfpranta** [comment]
Hello @rebornplusplus ,can you check for any flaws that I can mitigate next.And also please do check for the recursive formula if I have written it correctly. Besides, I wanted to avoid writing the recursive formula since it basically tells the reader what the answer is before he goes to see the implementation part. And I thought simulating through a test case might help the reader find a pattern/logic which can help them think about implementing the problem even though my writing seems jeopardized and out of the world

File: 1011/en.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# LOJ 1011 - Marriage Ceremonies
+
+In this problem, you will be given `T` testcases. The first line of each test case contains  integer `N` which is the number of both man and woman. Now each of the N lines line consist of N integers denoting the priority of the jth woman toward ith man.Let's analyze this with a testcase
+
+3
+1 2 3
+6 5 4
+8 1 2
+
```
Comment (Tahanima) [inline 1011/en.md:9]: Wrap this around 3 backticks. Like this:
```
3
1 2 3
6 5 4
8 1 2
```

File: 1011/en.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+And then he got married with the 2nd woman.So upto now 3rd man married 1st woman,2nd man married 2nd woman and now the only man and woman left are 1st and 3rd respectively and they got engaged.
+Now that sum of the priority is 3+5+8=16.And this combination is the maximum of all the combinations without having any overlappings and that's why this is the answer.
+
+### Approach:
+In order to solve this problem, there are a few prerequisites that need to be fulfilled which are: you need to have a basic understanding of dynamic programming and bit manipulations. The tutorials you can look and follow to develop your knowledge regarding these topics is provided in the "Resources" section
+
+To solve this problem we will make a "Bitmask funtion" where the parameters that need to be taken are the `ith` man and current mask which is under effect currently.
+In order to invoke this function we will send (0.0) and start computing the answer of this problem.Now for the `ith` man we will try to look for `1<<j` woman where `1<<j` can be 1,2,4
```
Comment (Tahanima) [inline 1011/en.md:22]: Here, (0.0) should be (0, 0).

File: 1011/en.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#include<bits/stdc++.h>
+using namespace std;
+int a[16][16];
+int dp[1<<16];//1<<16==2^16 since in worst case there can be 16 marriage ceremonies
+int n;
+int bitmask(int i,int mask)
+{
+    if(mask==(1<<n)-1) //base case
+        return 0;
+    int &ans=dp[mask];//& is a reference operator and any change brought to a reference variable will be set to the dp array at mask index because of this operator
```
Comment (Tahanima) [inline 1011/en.md:55]: The explanation regarding the reference operator is unnecessary here. So, please remove it.

File: 1011/en.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+### Resources:
+Here are some resources of  **Bit Manipulation,Basic Dp and BitMask Dp** so that you can have a deeper understanding and learn to solve problems using these resources:-
+
+- [Shafaetsplanet](https://www.shafaetsplanet.com/?p=1357) **(BitMask Dp)**
+- [HackerEarth](https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/tutorial/) **(Basics of Bit Manipulation)**
+- [Bangladesh Advanced Computing Society - BACS](https://www.youtube.com/watch?v=cbgdSX2pXcQ)   **(BitMask Dp)**
+- [GeeksforGeeks](https://www.geeksforgeeks.org/dynamic-programming/)   **(Basic and Advanced Types of DP)**
+
+
+### C++ Implementation:
```
Comment (Tahanima) [inline 1011/en.md:44]: Use the below format:
### C++
-----

File: 1011/en.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+Now since 1st woman and 2nd woman are taken the mask is 3==011 . Now for 3rd woman we can't take the either "1s" of the binary number.So the only option left for 3rd man is the 3rd woman which is the 3rd bit from the right which is still zero
+Now choosing the 3rd bit that is the 3rd woman we make `mask|(1<<2)==3|4==7` that is, we considered all the man and woman without any overlapping and so we completed a combination and so we return 0.And how we ignored the overlapping?For the ith man we simply avoided woman or bit which is set as `1` and we condered the bits which are `0` and considered all combinations we can take and stored the maximum inside a variable
+That maximum number which is a combination is the answer.The number of total combinations is `2^N-1` and we will have to compute all of them and return the maximum one out there which will denote the maximum sum of priority.
+
+The recursive formula of this problem is:
+`
+ bitmask(i,(1<<n)-1)=0 
+ bitmask(i,mask)=max(a[i][j]+bitmask(i+1,mask|(1<<j)) where i<=n,j<=n and a[i][j] consists of priority of ith man for jth woman
+`
```
Comment (Tahanima) [inline 1011/en.md:33]: Wrap this around 3 backticks.

**Review by Tahanima — COMMENTED** [review]
@nfpranta Please fix all the requested changes. Also, there are a few grammatical errors in your writing for example, `women` and not `womans`. Please fix those. In a DP tutorial, I would prefer that it is clearly explained what made you come to the conclusion that it can be solved by DP and why Bitmask DP? Your writing is missing that part. And, lastly a small feedback. Use proper spacing wherever required, like for example `1 << j` instead of `1<<j`.

File: 1011/en.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+### Resources:
+Here are some resources of  **Bit Manipulation,Basic Dp and BitMask Dp** so that you can have a deeper understanding and learn to solve problems using these resources:-
+
+- [Shafaetsplanet](https://www.shafaetsplanet.com/?p=1357) **(BitMask Dp)**
+- [HackerEarth](https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/tutorial/) **(Basics of Bit Manipulation)**
+- [Bangladesh Advanced Computing Society - BACS](https://www.youtube.com/watch?v=cbgdSX2pXcQ)   **(BitMask Dp)**
+- [GeeksforGeeks](https://www.geeksforgeeks.org/dynamic-programming/)   **(Basic and Advanced Types of DP)**
+
+
```
Comment (Tahanima) [inline 1011/en.md:43]: Pease remove this unnecessary blank line.

**Comment by nfpranta** [comment]
Hello @Tahanima apu.Can you please have a look into this now?

**Comment by faiyaz26** [comment]
Thanks for your contribution

**Pull request merged**